test/src/test/java/jenkins/security/ResourceDomainTest.java

```diff
@@ -0,0 +1,281 @@
+package jenkins.security;
+
+import com.gargoylesoftware.htmlunit.Page;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import hudson.ExtensionList;
+import hudson.model.DirectoryBrowserSupport;
+import hudson.model.FreeStyleProject;
+import hudson.model.Item;
+import jenkins.model.Jenkins;
+import jenkins.model.JenkinsLocationConfiguration;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.CreateFileBuilder;
+import org.jvnet.hudson.test.For;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.MockAuthorizationStrategy;
+
+import java.net.URL;
+import java.util.UUID;
+
+@Issue("JENKINS-41891")
+@For({ ResourceDomainRootAction.class, ResourceDomainFilter.class, ResourceDomainConfiguration.class })
+public class ResourceDomainTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private static final String RESOURCE_DOMAIN = "127.0.0.1";
+
+    @Before
+    public void prepare() throws Exception {
+        String resourceRoot;
+        URL root = j.getURL(); // which always will use "localhost", see JenkinsRule#getURL()
```

Comment:
Assert that to be safe.

Reply:
Sure, but all of this will break anyway if the replacement doesn't work since it'll be the same URL.